invenio_rdm_records/records/systemfields/draft_status.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Draft status field.
+
+The DraftStatus is used to calculate the status of the draft based on its
+associated review request.
+
+For instance:
+
+.. code-block:: python
+
+    class Record():
+        draft_status = DraftStatus()
+
+"""
+
+from flask_babelex import gettext as _
+from invenio_records.systemfields import SystemField
+
+from invenio_rdm_records.services.errors import ReviewStateError
+
+
+class DraftStatus(SystemField):
+    """Draft status field which checks the `parent.review` state."""
+
+    available_review_to_draft_status = dict(
+        draft=_("Draft"),
```

Comment:
Question: This one is a bit tricky. The "list uploads" page needs to display a facet over this status. The facet I think should display:

- Draft: draft + draft with review states 
- In review
- Review expired 
- Review declined
- Published

This assumes end-users won't understand the difference between ``Draft`` and ``Draft with review``. If we think end-users will understand, and/or that users want to find all drafts being submitted for review then we can leave as is.